File: src/main/java/org/example/entity/Category.java
package org.example.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.example.enums.CategoryType;

import java.util.UUID;

@Entity
@Table(name = "categories")
@Getter
@Setter
public class Category {

    @Id
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    private String name;

    @Enumerated(EnumType.STRING)
    private CategoryType type;

    private boolean isSystem;
}

File: frontend/src/types/category.ts
// Тип категории: доход или расход.
export type CategoryType = 'INCOME' | 'EXPENSE';

// Модель категории, используемая в приложении.
export interface Category {
    // Уникальный идентификатор категории.
    id: string;
    // Отображаемое название категории.
    name: string;
    // Тип категории (доход/расход).
    type: CategoryType;
    // Признак системной (предустановленной) категории.
    system: boolean;
}

// Данные для создания новой категории.
export interface CategoryCreatePayload {
    // Название новой категории.
    name: string;
    // Тип создаваемой категории.
    type: CategoryType;
}
